azure/functions/decorators/function_app.py

```diff
@@ -1620,8 +1625,15 @@ def get_functions(self) -> List[Function]:
 
         :return: List of functions in the function app.
         """
-        return [function_builder.build(self.auth_level) for function_builder
-                in self._function_builders]
+        functions = [function_builder.build(self.auth_level)
+                     for function_builder in self._function_builders]
+
+        if all(not function.is_http_function() for function in functions):
```

Comment:
Could the trigger type be checked for all, rather than adding a flag?
or a flag at the function all level that is set whenever HTTP trigger is added (to not do a loop at all)?

Reply:
Dont see good way to put in function app level though i know it helps performance wise. function does not have access to functionapp and if it could, will create tight coupling as functionapp already have access to functions. 

get_functions is internally used and only once so it is not that bad. I am thinking to add a new flag in functionapp and only update the flag if no http functions found.
<img width="370" alt="image" src="https://user-images.githubusercontent.com/12449837/180883526-fa50d469-9b1c-4fe7-a09d-637c023fe991.png">